src/entity/Living.php

```diff
@@ -416,6 +416,10 @@ public function applyDamageModifiers(EntityDamageEvent $source) : void{
 		$source->setModifier(-$source->getFinalDamage() * min(ceil(min($totalEpf, 25) * (mt_rand(50, 100) / 100)), 20) * 0.04, EntityDamageEvent::MODIFIER_ARMOR_ENCHANTMENTS);
 
 		$source->setModifier(-min($this->getAbsorption(), $source->getFinalDamage()), EntityDamageEvent::MODIFIER_ABSORPTION);
+
+		if($cause === EntityDamageEvent::CAUSE_FALLING_BLOCK && $this->armorInventory->getHelmet() instanceof Armor){
+			$source->setModifier(-min($source->getFinalDamage() / 3, 10), EntityDamageEvent::MODIFIER_HARD_HELMET);
```

Comment:
Is there a reason why we can't just throw this under MODIFIER_ARMOR?

Reply:
Mainly because it makes it easier to identify when extra damage should be applied to the helmet and `MODIFIER_ARMOR` is only used if the cause do `EntityDamageEvent::canBeReducedByArmor()` return `true` and any armor reduces damage.

Reply:
Wouldn't a combination of CAUSE_FALLING_BLOCK and MODIFIER_ARMOR achieve the same?

Reply:
Never mind, I suppose this way is less fragile.

Reply:
My research into the current version of Bedrock suggests that this is reduced by 25%, same as Java.

Reply:
Why is this marked as resolved?

Reply:
> My research into the current version of Bedrock suggests that this is reduced by 25%, same as Java.

done.